src/tests/ftest/util/ec_utils.py

```diff
@@ -133,3 +142,167 @@ def ior_read_dataset(self, parity=1):
+                    self.ec_container_create(cont_count, oclass[0])
+                    self.single_type_param_update(cont_count, sizes)
+                    # Write the data
+                    self.container[cont_count].write_objects(
+                        obj_class=oclass[0])
+                    cont_count += 1
+                    if results is not None:
+                        results.put("PASS")
+                except (CommandFailure, DaosApiError, DaosTestError) as _error:
+                    if results is not None:
+                        results.put("FAIL")
+                    raise
+
+    def read_single_type_dataset(self, results=None, parity=1):
+        """Read single type data and verify for different EC object
+            and different sizes.
+
+        Args:
+           parity(int): object parity number for reading, default All.
+        """
+        cont_count = 0
+        for oclass in self.obj_class:
+            for _sizes in  self.singledata_set:
+                # Skip the object type if server count does not meet
+                # the minimum EC object server count
+                if oclass[1] > self.server_count:
+                    continue
+                parity_set = "P{}".format(parity)
+                # Read the requested data+parity data set only
+                if parity != 1 and parity_set not in oclass[0]:
+                    print("Skipping Read as object type is {}"
+                          .format(oclass[0]))
+                    cont_count += 1
+                    continue
+                # Read data and verified the content
+                try:
+                    if not self.container[cont_count].read_objects():
+                        self.fail("Data verification Error")
+                        if results is not None:
+                            results.put("FAIL")
+                    cont_count += 1
+                    if results is not None:
+                        results.put("PASS")
+                except (CommandFailure, DaosApiError, DaosTestError) as _error:
+                    if results is not None:
+                        results.put("FAIL")
+                    raise
+
+    def start_online_single_operation(self, operation, parity=1):
+        """Do Write/Read operation with single data type.
+
+        Args:
+            operation (string): Write/Read operation
+        """
+        # Create the single data Write/Read threads
+        if operation == 'WRITE':
+            job = threading.Thread(target=self.write_single_type_dataset,
+                                   kwargs={"results": self.out_queue})
+        elif operation == 'READ':
+            job = threading.Thread(target=self.read_single_type_dataset,
+                                   kwargs={"results": self.out_queue,
+                                           "parity": parity})
+
+        # Launch the IOR thread
+        job.start()
+
+        # Kill the server rank while IOR in progress
```

Comment:
Minor: May be need to remove IOR from comment?

Reply:
yes will fix it.